src/core/init/duration_updater.ts

```diff
@@ -58,9 +63,20 @@ export default function DurationUpdater(
       if (isNaN(mediaSource.duration) || !isFinite(mediaSource.duration) ||
           Math.abs(mediaSource.duration - newDuration) > 0.01)
       {
```

Comment:
Aren't we entering this if because when the `MediaSource` is not `"open"` the `duration` is set to `0`? In that case this is not the "real" duration and this check is both invalid and unnecessary, no?

For me the most logical way would be to either:
  - Check before that `if` if the MediaSource's `readyState` is open.
    We may duplicate that same `if` here (in the synchronous and asynchronous case), or maybe create an `Observable.of(null)` in the synchronous case but that's not the most beautiful code
  - only listen to manifestUpdate events - to trigger that logic - when the MediaSource is open (closer to how I think of that problem - but maybe not how you/everyone else thinks!). This leads basically to reversing the Observable and its switchMap (first listening to the MediaSource's state, then switchMapping to the Manifest's `manifestUpdate` events).
    However it might be more complex to implement (because we may need to catch other readyState` transitions)